src/wallet/mod.rs

```diff
@@ -100,11 +99,13 @@ impl WalletManager {
         match password {
             None => {
                 let store = self.load_unlocked().await?;
-                return Ok(store
+                let wallet = store
                     .wallets
                     .get(&reference)
-                    .ok_or_else(|| anyhow!("Wallet reference not found: {:?}", &reference))?
-                    .to_owned())
+                    .ok_or_else(|| anyhow!("Wallet reference not found: {:?}", &reference))
+                    .unwrap() // we want to catch the error if the wallet id doesn't exist
```

Comment:
It would be nicer if the error was passed to the caller? So that in the future we can customize the way errors are shown, etc, rather than relying on the rust panic message.